src/main/java/org/influxdb/InfluxDBFactory.java

```diff
@@ -103,6 +109,25 @@ public static InfluxDB connect(final String url, final String username, final St
     Preconditions.checkNonEmptyString(url, "url");
     Preconditions.checkNonEmptyString(username, "username");
     Objects.requireNonNull(client, "client");
+    checkUrl(url);
     return new InfluxDBImpl(url, username, password, client, responseFormat);
   }
+
+  /**
+   * Check url is legal
+   */
+  private static void checkUrl(final String url) {
```

Comment:
method should also be part of Preconditions

Reply:
got it